src/components/VideoCard/VideoCard.jsx

```diff
@@ -7,19 +7,49 @@ import {
   BsSave2,
   BsHandThumbsUp,
 } from "react-icons/bs";
-import { getViews } from "../../utils/getViews";
+import { getViews } from "../../utils";
+import { useData } from "../../context";
 
-export const VideoCard = ({
-  thumbnail,
-  name,
-  actors,
-  category,
-  language,
-  releaseYear,
-  duration,
-  views,
-}) => {
+export const VideoCard = (video) => {
+  const {
+    _id,
+    thumbnail,
+    name,
+    actors,
+    category,
+    language,
+    releaseYear,
+    duration,
+    views,
+  } = video;
   const [menu, toggleMenu] = useState(false);
+  const { watchLater, likedVideos, dispatch } = useData();
+
+  const isInWatchLater = watchLater.some((vid) => vid._id === _id);
+  const isInLikedVideos = likedVideos.some((vid) => vid._id === _id);
+
+  const contextMenuHandler = (e) => {
+    switch (e.target.innerText) {
```

Comment:
use e.target.id for it rather than matching it with the help of innerText . Because if later on you have made any changes in text then you need to change it here as well.

Reply:
Nice suggestion, will try to implement it.